chandra_aca/dark_subtract.py

```diff
@@ -0,0 +1,431 @@
+
+
+@lru_cache(maxsize=2)
+def get_mica_images(start: CxoTimeLike, stop: CxoTimeLike, cols=None):
+    """
+    Get ACA images from mica for a given time range.
+
+    These mica images are centered in the 8x8 images via the centered_8x8=True option
+    to aca_l0.get_slot_data to match the maude images.
+
+    Parameters
+    ----------
+    start : CxoTimeLike
+        Start time of the time range.
+    stop : CxoTimeLike
+        Stop time of the time range.
+
+    Returns
+    -------
+    dict
+        dict with a astropy.table of ACA image data for each slot.
+
+    """
+    if cols is None:
+        cols = [
+            "TIME",
+            "QUALITY",
+            "IMGFUNC1",
+            "IMGRAW",
+            "IMGSIZE",
+            "IMGROW0",
+            "IMGCOL0",
+            "BGDAVG",
+            "INTEG",
+        ]
+    else:
+        for col in ["TIME", "IMGRAW", "IMGROW0", "IMGCOL0"]:
+            if col not in cols:
+                cols.append(col)
+
+    slot_data = {}
+    for slot in range(8):
+        slot_data[slot] = aca_l0.get_slot_data(
+            start,
+            stop,
+            imgsize=[4, 6, 8],
+            slot=slot,
+            columns=cols,
+            centered_8x8=True,
+        )
+        slot_data[slot] = Table(slot_data[slot])
+
+        # Rename and rejigger colums to match maude
+        IMGROW0_8X8 = slot_data[slot]["IMGROW0"].copy()
+        IMGCOL0_8X8 = slot_data[slot]["IMGCOL0"].copy()
+        IMGROW0_8X8[slot_data[slot]["IMGSIZE"] == 6] -= 1
+        IMGCOL0_8X8[slot_data[slot]["IMGSIZE"] == 6] -= 1
+        IMGROW0_8X8[slot_data[slot]["IMGSIZE"] == 4] -= 2
+        IMGCOL0_8X8[slot_data[slot]["IMGSIZE"] == 4] -= 2
+        slot_data[slot]["IMGROW0_8X8"] = IMGROW0_8X8
+        slot_data[slot]["IMGCOL0_8X8"] = IMGCOL0_8X8
+        slot_data[slot].rename_column("IMGRAW", "IMG")
+
+    return slot_data
+
+
+@retry.retry(exceptions=requests.exceptions.RequestException, delay=5, tries=3)
+@lru_cache(maxsize=2)
+def get_maude_images(start: CxoTimeLike, stop: CxoTimeLike, fetch_limit_hours=100):
+    """
+    Get ACA images from MAUDE for a given time range.
+
+    Parameters
+    ----------
+    start : CxoTimeLike
+        Start time of the time range.
+    stop : CxoTimeLike
+        Stop time of the time range.
+    fetch_limit_hours : int
+        Maximum number of hours to fetch from MAUDE in one go.
+
+    Returns
+    -------
+    dict
+        dict with a astropy.table of ACA image data for each slot.
+    """
+    tstart = CxoTime(start).secs
+    tstop = CxoTime(stop).secs
+
+    if tstop - tstart > fetch_limit_hours * 60 * 60:
+        raise ValueError("Time range too large for maude fetch, limit is 100 hours")
+
+    slot_chunks = []
+    # Break maude fetches into max 3 hour chunks required by maude_decom fetch
+    for mstart in np.arange(tstart, tstop, 60 * 60 * 3):
+        mstop = np.min([tstop, mstart + 60 * 60 * 3])
+        # Add a retry here if necessary
```

Comment:
Leftover comment - I stuck the retry in as a decorator but it is basically untested.